File: 0218.py
def digital_root(n):
    return n if n < 10 else digital_root(sum(int(digit) for digit in str(n)))

def digital_root2(n):
    return n if n == 0 else (n - 1) % 9 + 1

print(digital_root(78))
print(digital_root(12))
print(digital_root(3))
print(digital_root2(78))
print(digital_root2(12))
print(digital_root2(3))

File: 0218.rb
def digital_root(n)
  n < 10 ? n : digital_root(n.digits.sum)
end

def digital_root2(n)
  n.zero? ? 0 : (n - 1) % 9 + 1
end

puts digital_root(16)
puts digital_root(942)
puts digital_root(132189)
puts digital_root(493193)
